src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/TeleporterSystem.java

```diff
@@ -57,8 +46,35 @@
     @ReceiveEvent(components = {SetTeamOnActivateComponent.class})
     public void onActivate(ActivateEvent event, EntityRef entity) {
         EntityRef player = event.getInstigator();
-        String team = setPlayerTeamToTeleporterTeam(player, entity);
-        handlePlayerTeleport(player, team);
+        if (properTeamSize(entity, player)) {
+            String team = setPlayerTeamToTeleporterTeam(player, entity);
+            handlePlayerTeleport(player, team);
+        }
+    }
+
+    private boolean properTeamSize(EntityRef teleporter, EntityRef player) {
```

Comment:
It's recommended to prefix boolean variables and functions with "is", so that it's clear what the respective `true` and `false` case will indicate.

```suggestion
    private boolean isProperTeamSize(EntityRef teleporter, EntityRef player) {
```

Reply:
Done